### ISSUE TITLE
EmailReadImap trigger does not recognise inline attachment

### ISSUE THREAD

**Issue by terenaa** [n8n-io/n8n#689]
**Describe the bug**
EmailReadImap trigger does not recognise attachments when Content-Disposition is set to "inline". It is default behaviour of MacOS Mail application.

**To Reproduce**
Steps to reproduce the behaviour:
1. Send email with attachment using MacOS Mail application (v13.4) and run EmailReadImap trigger.

**Expected behaviour**
EmailReadImap trigger recognises attachments with Content-Disposition type of "attachment" and "inline".

**Example headers of email sent from the Mail application**
```
Return-Path: <xxxxx@gmail.com>
Received: from [0.0.0.0] ([0.0.0.0])
        by smtp.gmail.com with ESMTPSA id h15sm2928181ljk.24.2020.06.22.16.16.41
        for <xxxxx@gmail.com>
        (version=TLS1_2 cipher=ECDHE-ECDSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 22 Jun 2020 16:16:41 -0700 (PDT)
From: Krzysztof Janda <xxxxx@gmail.com>
Content-Type: multipart/alternative; boundary="Apple-Mail=_3A3B1B5E-A1C1-43BD-84F7-EF3D16200148"
Mime-Version: 1.0 (Mac OS X Mail 13.4 \(3608.80.23.2.2\))
Subject: Test
Message-Id: <6A3FB35D-DD64-4B66-8787-62F3AA339853@gmail.com>
Date: Tue, 23 Jun 2020 01:16:37 +0200
To: Krzysztof Janda <xxxxx@gmail.com>
X-Mailer: Apple Mail (2.3608.80.23.2.2)

--Apple-Mail=_3A3B1B5E-A1C1-43BD-84F7-EF3D16200148
Content-Transfer-Encoding: quoted-printable
Content-Type: text/plain; charset=utf-8

Test


--Apple-Mail=_3A3B1B5E-A1C1-43BD-84F7-EF3D16200148
Content-Type: multipart/mixed; boundary="Apple-Mail=_636B0FA7-6526-4173-9C2D-84324F882BAB"

--Apple-Mail=_636B0FA7-6526-4173-9C2D-84324F882BAB
Content-Transfer-Encoding: quoted-printable
Content-Type: text/html; charset=utf-8

<html><head><meta http-equiv=3D"Content-Type" content=3D"text/html; charset=
=3Dutf-8"></head><body style=3D"word-wrap: break-word; -webkit-nbsp-mode: s=
pace; line-break: after-white-space;" class=3D"">Test<div class=3D""><br class=3D""></div><div class=3D""><=
/div></body></html>
--Apple-Mail=_636B0FA7-6526-4173-9C2D-84324F882BAB
Content-Disposition: inline; filename=test.pdf
Content-Type: application/pdf; x-unix-mode=0644; name="test.pdf"
Content-Transfer-Encoding: base64


--Apple-Mail=_636B0FA7-6526-4173-9C2D-84324F882BAB
Content-Transfer-Encoding: 7bit
Content-Type: text/html; charset=us-ascii

<html><head><meta http-equiv="Content-Type" content="text/html; charset=us-ascii"></head><body style="word-wrap: break-word; -webkit-nbsp-mode: space; line-break: after-white-space;" class=""><div class=""></div></body></html>
--Apple-Mail=_636B0FA7-6526-4173-9C2D-84324F882BAB--
--Apple-Mail=_3A3B1B5E-A1C1-43BD-84F7-EF3D16200148--
```

**Comment by RicardoE105** [comment]
@terenaa thanks for putting that together. I will try to reproduce it as soon as I have time. I will keep you posted.

**Comment by pcross616** [comment]
I am running in to something similar but different and I wonder if it is related.  When some emails are sent and triggered on that haven an attachment (not inline, I have confirmed in the message source). I do not get any binary data.  If I forward the message to the inbox that the trigger is watching on it works fine.  Again comparing message source same disposition and content type.  The only thing I see is how the content-type is not escaped in some case when it does not work.  See below here.

This one does not work and you can see the content-type is not quoted around the boundaries.  You can also see there is no binary data at the top.

![Screen Shot 2020-08-15 at 4 22 49 PM](https://user-images.githubusercontent.com/141865/90323254-b27b0200-df13-11ea-832f-0d00d8d51c65.png)

When I forward that message from my mail client (not changing anything).  I get a tad different content-type and I get binary data.

![Screen Shot 2020-08-15 at 4 16 45 PM](https://user-images.githubusercontent.com/141865/90323263-d9d1cf00-df13-11ea-8f27-95c20fe01020.png)

I dont have an easy way to see what the Node object thinks of the content-disposition in n8n, since its just binary text.  But comparing the two messages in Outlook as source they look the same.

This one on the left is the original email that did not work in n8n and the one on the right is the email I forwarded and worked in n8n.
![Screen Shot 2020-08-15 at 4 26 36 PM](https://user-images.githubusercontent.com/141865/90323297-56fd4400-df14-11ea-83b3-8981ca476246.png)

I am trying to debug this more but there is not a lot different when it works and it doesn't except for the content-type escaping.

Thanks.

**Comment by Rupenieks** [comment]
Hey @terenaa, we've implemented a solution for this issue. It'll be released as soon as possible, I will update you here once that happens.

FYI: If expecting emails with content type: inline, you will have to set the format setting to "resolved".

**Comment by RicardoE105** [comment]
@pcross616

**Comment by janober** [comment]
Just got released with n8n@0.79.0
